### ISSUE TITLE
First own language server approach

### ISSUE THREAD

**Issue by katauber** [metafacture/metafacture-lsp#1]
After the from Xtext generated language server for Metafacture Fix and Flux is very simple, we want a language server which supports some more features. So we need to implement our own language server.
There language server protocol bases like [lsp4clj](https://github.com/clojure-lsp/lsp4clj) or [lsp4j](https://github.com/eclipse-lsp4j/lsp4j) which I think we should use. I prefer clojure as always :)

First we thought a good start could be to reimplement the function of the existing Fix language server.
We first need the parser of our language, so the first step could be moving from [Xtext to ANTLR](https://github.com/metafacture/metafacture-fix/pull/298) for Metafacture Fix.
@fsteeg and I discussed offline that we also could start with a language server for Metafactre Flux, because Metafacture Flux in Metafacture Core already uses ANTLR.
So the first step is to implement a minimal language server for Metafacture Flux with [lsp4clj](https://github.com/clojure-lsp/lsp4clj) or [lsp4j](https://github.com/eclipse-lsp4j/lsp4j)

**Comment by fsteeg** [comment]
We could start with auto-completion for Flux commands based on the `flux-commands.properties` files, e.g. https://github.com/metafacture/metafacture-core/blob/master/metafacture-biblio/src/main/resources/flux-commands.properties

**Comment by blackwinter** [comment]
> We could start with auto-completion for Flux commands based on the `flux-commands.properties` files

Hm, as alluded to in metafacture/metafacture-core#706, I'd rather see us get rid of these (redundant) mappings and instead rely solely on the annotations. If you're going to depend on these properties files, though, then that's no longer an option. OTOH, what's your plan for Fix commands? Since there is no analogous `fix-commands.properties` (it only contains package names, not the actual commands), you'd have to use a different approach anyway, wouldn't you?

**Comment by fsteeg** [comment]
Yes, right, thanks for bringing that up, we should use the annotations, as we'll do in Fix.

@katauber: for initial tests of setup and client-server interaction, we might still use one of the files, or some test values.
